Tests/SwiftKueryORMTests/TestSave.swift

```diff
@@ -17,20 +50,48 @@ class TestSave: XCTestCase {
         var name: String
         var age: Int
     }
-
+    /**
+      Testing that the correct SQL Query is created to save a Model
+    */
     func testSave() {
         let connection: TestConnection = createConnection()
         Database.default = Database(single: connection)
         performTest(asyncTasks: { expectation in
             let person = Person(name: "Joe", age: 38)
             person.save { p, error in
                 XCTAssertNil(error, "Save Failed: \(String(describing: error))")
-                XCTAssertNotNil(connection.query, "Find Failed: Query is nil")
+                XCTAssertNotNil(connection.query, "Save Failed: Query is nil")
                 if let query = connection.query {
-                  let expectedQuery1 = "INSERT INTO People (name, age) VALUES ('Joe', 38)"
-                  let expectedQuery2 = "INSERT INTO People (age, name) VALUES (38, 'Joe')"
+                  let expectedPrefix = "INSERT INTO People"
+                  let expectedSQLStatement = "VALUES"
+                  let expectedTuples = [("name", "Joe"), ("age", "38")]
+
                   let resultQuery = connection.descriptionOf(query: query)
-                  XCTAssert(resultQuery == expectedQuery1 || resultQuery == expectedQuery2)
+                  //Regex to extract the columns and values of an insert
+                  //statement, such as:
+                  //INSERT into table (columns) VALUES (values)
+                  let regexPattern = ".*\\((.*)\\)[^\\(\\)]*\\((.*)\\)"
+                  let groups = resultQuery.capturedGroups(withRegex: regexPattern)
+                  XCTAssertEqual(groups.count, 2)
+
+                  // Extracting the columns and values from the captured groups
+                  let columns = groups[0].filter { $0 != " " }.split(separator: ",")
+                  let values = groups[1].filter { $0 != " " && $0 != "'" }.split(separator: ",")
+
+                  // Creating the result tuples (Column, Value)
+                  var resultTuples: [(String, String)] = []
+                  for (column, value) in zip(columns, values) {
+                    resultTuples.append((String(column), String(value)))
+                  }
+
+                  // Asserting the results which the expectations
+                  XCTAssertTrue(resultQuery.hasPrefix(expectedPrefix))
+                  XCTAssertTrue(resultQuery.contains(expectedSQLStatement))
+                  XCTAssertEqual(resultTuples.count, expectedTuples.count)
+                  for i in 0..<expectedTuples.count {
+                    XCTAssertEqual(resultTuples[i].0, expectedTuples[i].0)
+                    XCTAssertEqual(resultTuples[i].1, expectedTuples[i].1)
+                  }
```

Comment:
Thanks for taking the time to code this up.  I fear we've turned a slightly annoying but simple thing into a clever and hideous thing, but it does mean this approach would scale to an arbitrary number of fields.  If we use this code for another test in the future we should extract it as a utility function.